File: bugsnag-android-core/src/androidTest/java/com/bugsnag/android/EventPayloadTest.java
package com.bugsnag.android;

import static com.bugsnag.android.BugsnagTestUtils.generateAppWithState;
import static com.bugsnag.android.BugsnagTestUtils.generateDeviceWithState;
import static com.bugsnag.android.BugsnagTestUtils.streamableToJson;
import static org.junit.Assert.assertEquals;

import com.bugsnag.android.internal.ImmutableConfig;

import androidx.test.filters.SmallTest;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

@SmallTest
public class EventPayloadTest {

    private EventPayload eventPayload;

    /**
     * Generates a eventPayload
     */
    @Before
    public void setUp() {
        ImmutableConfig config = BugsnagTestUtils.generateImmutableConfig();
        RuntimeException exception = new RuntimeException("Something broke");
        SeverityReason severityReason = SeverityReason.newInstance(SeverityReason.REASON_ANR);
        Event event = new Event(exception, config, severityReason, NoopLogger.INSTANCE);
        event.setApp(generateAppWithState());
        event.setDevice(generateDeviceWithState());
        eventPayload = new EventPayload("api-key", event, new Notifier(), config);
    }

    @Test
    public void testInMemoryError() throws JSONException, IOException {
        JSONObject reportJson = streamableToJson(eventPayload);
        assertEquals(1, reportJson.getJSONArray("events").length());
    }

    @Test
    public void testModifyingGroupingHash() throws JSONException, IOException {
        String groupingHash = "File.java:300429";
        eventPayload.getEvent().setGroupingHash(groupingHash);

        JSONObject reportJson = streamableToJson(eventPayload);
        JSONArray events = reportJson.getJSONArray("events");
        JSONObject event = events.getJSONObject(0);
        assertEquals(groupingHash, event.getString("groupingHash"));
    }

}

File: bugsnag-android-core/src/test/java/com/bugsnag/android/EventPayloadTest.kt
package com.bugsnag.android

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotSame
import org.junit.Test

class EventPayloadTest {

    @Test
    fun testCloneNotifier() {
        val original = Notifier()
        val config = BugsnagTestUtils.generateImmutableConfig()
        val payload = EventPayload("api-key", null, null, original, config)
        val copy = payload.notifier
        assertNotSame(original, copy)
        assertNotSame(original.dependencies, copy.dependencies)
        assertEquals(original.dependencies, copy.dependencies)
        assertEquals(original.name, copy.name)
        assertEquals(original.url, copy.url)
        assertEquals(original.version, copy.version)
    }
}
